codegen/src/main/java/software/amazon/awssdk/codegen/emitters/tasks/PaginatorsGeneratorTasks.java

```diff
@@ -45,14 +47,30 @@ protected boolean hasTasks() {
     @Override
     protected List<GeneratorTask> createTasks() throws Exception {
         info("Emitting paginator classes");
+        return Stream.concat(createSyncTasks(), createASyncTasks())
+                     .collect(Collectors.toList());
+    }
+
+    private Stream<GeneratorTask> createSyncTasks() {
```

Comment:
You should be able to reduce a little duplication here and create a method that takes in this::createSyncTask this::createAsyncTask respectively.